subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/AbstractOptionElement.java

```diff
@@ -32,7 +33,7 @@
     private final Class<?> optionType;
     private final NotationParser<CharSequence, ?> notationParser;
 
-    public AbstractOptionElement(String optionName, Option option, Class<?> optionType, Class<?> declaringClass, NotationParser<CharSequence, ?> notationParser) {
+    AbstractOptionElement(String optionName, Option option, Class<?> optionType, Class<?> declaringClass, NotationParser<CharSequence, ?> notationParser) {
```

Comment:
What's the main reason for making things package visible and final ? Is it to prevent end users from accidentally using internal APIs?